centreon/www/include/eventLogs/xml/data.php

```diff
@@ -291,29 +262,51 @@
 }
 
 if ($StartDate != "") {
-    preg_match("/^([0-9]*)\/([0-9]*)\/([0-9]*)/", $StartDate, $matchesD);
-    preg_match("/^([0-9]*):([0-9]*)/", $StartTime, $matchesT);
-    $start = mktime($matchesT[1], $matchesT[2], "0", $matchesD[1], $matchesD[2], $matchesD[3]);
+    $dateTime = DateTime::createFromFormat('m/d/Y H:i', "$StartDate $StartTime");
+    if ($dateTime !== false) {
+        $start = $dateTime->getTimestamp();
+    } else {
+        $centreonLog::create()->error(
+            CentreonLog::TYPE_BUSINESS_LOG,
+            "Invalid date format: $StartDate $StartTime"
+        );
+    }
 }
 if ($EndDate != "") {
-    preg_match("/^([0-9]*)\/([0-9]*)\/([0-9]*)/", $EndDate, $matchesD);
-    preg_match("/^([0-9]*):([0-9]*)/", $EndTime, $matchesT);
-    $end = mktime($matchesT[1], $matchesT[2], "0", $matchesD[1], $matchesD[2], $matchesD[3]);
+    $dateTime = DateTime::createFromFormat('m/d/Y H:i', "$EndDate $EndTime");
+    if ($dateTime !== false) {
+        $end = $dateTime->getTimestamp();
+    } else {
+        $centreonLog->error(
+            CentreonLog::TYPE_BUSINESS_LOG,
+            "Invalid date format: $EndDate $EndTime"
+        );
+    }
 }
 
-// setting the startDate/Time using the user's chosen period
-// and checking if the start date/time was set by the user, to avoid to display/export the whole data since 1/1/1970
+// Setting the startDate/Time using the user's chosen period
 $period = 86400;
 if ($auto_period > 0 || $start === 0) {
     $period = (int)$auto_period;
```

Comment:
This variable is already of type Integer
```suggestion
    $period = $auto_period;
```